### PR TITLE
Avoid the log entry with the ModuleAlreadyExists exception when enabl…

### LINKED ISSUE #16103: ModuleAlreadyExistsException on app activation
```json


{
  "reqId": "tzEThQXcmCZWF5j5rc7y",
  "remoteAddr": "192.168.178.24",
  "app": "index",
  "message": "Exception: {\"Exception\":\"OC\\\\Encryption\\\\Exceptions\\\\ModuleAlreadyExistsException\",\"Message\":\"Id \\\"OC_DEFAULT_MODULE\\\" already used by encryption module \\\"ownCloud Default Encryption\\\"\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/apps\\\/encryption\\\/appinfo\\\/application.php(109): OC\\\\Encryption\\\\Manager->registerEncryptionModule('OC_DEFAULT_MODU...', 'ownCloud Defaul...', Object(Closure))\\n#1 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/apps\\\/encryption\\\/appinfo\\\/app.php(30): OCA\\\\Encryption\\\\AppInfo\\\\Application->registerEncryptionModule()\\n#2 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/lib\\\/private\\\/app.php(388): require('\\\/home\\\/deepdiver...')\\n#3 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/settings\\\/ajax\\\/navigationdetect.php(30): OC_App::getAppNavigationEntries('encryption')\\n#4 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/lib\\\/private\\\/route\\\/route.php(153) : runtime-created function(1): require_once('\\\/home\\\/deepdiver...')\\n#5 [internal function]: __lambda_func(Array)\\n#6 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/lib\\\/private\\\/route\\\/router.php(272): call_user_func('\\\\x00lambda_2587', Array)\\n#7 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/lib\\\/base.php(860): OC\\\\Route\\\\Router->match('\\\/settings\\\/ajax\\\/...')\\n#8 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/index.php(40): OC::handleRequest()\\n#9 {main}\",\"File\":\"\\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core-autotest\\\/lib\\\/private\\\/encryption\\\/manager.php\",\"Line\":101}",
  "level": 4,
  "time": "2015-05-06T08:30:08+00:00",
  "method": "GET",
  "url": "\/index.php\/settings\/ajax\/navigationdetect.php?app=encryption"
}

### PR DESCRIPTION
**Pull request by nickvergessen**
…ing the app

Fix #16103

@DeepDiver1975 @schiesbn 

The only other idea would be to refactor how the enable app works:
#### old
1. Enable app
2. In case of success gather navigation data
3. In case of success add data from step2 to navigation
#### new
1. Gather navigation data
2. Enable app
3. In case of success add data from step1 to navigation

=> The risc of changing this is we might execute code of the app, before it is enabled, etc.

### CODE REVIEW

File: lib/private/app.php [hunk 1]
```diff
@@ -385,7 +385,13 @@ public static function setActiveNavigationEntry($id) {
 	public static function getAppNavigationEntries($app) {
 		if (is_file(self::getAppPath($app) . '/appinfo/app.php')) {
 			OC::$server->getNavigationManager()->clear();
-			require $app . '/appinfo/app.php';
+			try {
+				require $app . '/appinfo/app.php';
```
Comment (DeepDiver1975) [inline lib/private/app.php]: what about using require_once here instead? http://php.net/manual/de/function.require-once.php

Comment (nickvergessen) [inline lib/private/app.php]: That will not include the file anymore, so the app.php code will never be executed and no entry will ever be added ;)

Comment (DeepDiver1975) [inline lib/private/app.php]: > That will not include the file anymore, so the app.php code will never be executed and no entry will ever be added ;)

why - it should exactly be required once - which is what we want to have. Or do I miss anything here?

Comment (nickvergessen) [inline lib/private/app.php]: it was already included earlier, when the app was loaded (before we reach the code of the `ajax/navigationdetect.php`). We clear/reset the navigation manager here, then load the `app.php` file again and see whether a navigation entry has been registered.

Other ideas how to change this (not using app.php being one example) are at https://github.com/owncloud/core/issues/14881 :dont_open_that_can_of_worms_here_please: :broken_heart:

Comment (DeepDiver1975) [inline lib/private/app.php]: hmmmm .. bloody hell ...

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by LukasReschke** [comment]
:see_no_evil: :hear_no_evil: :speak_no_evil: 

:+1: 

:see_no_evil: :hear_no_evil: :speak_no_evil:

**Pull request merged**